zcbor/zcbor.py

```diff
@@ -1897,9 +1897,9 @@ def type_name(self):
         if self.multi_var_condition():
             name = self.raw_type_name()
             if self.val_type_name() == name:
-                name = name + "_"
+                name = name + "_m"
```

Comment:
What does `_m` indicate? Should we add a comment?

Reply:
Thanks, I found it wasn't needed, so I removed it.
Then I changed _d to _m. I also added a bunch of comments in generate_base_name()